dds/DCPS/DCPS_Utils.cpp

```diff
@@ -252,10 +252,21 @@ compatibleQOS(OpenDDS::DCPS::IncompatibleQosStatus* writerStatus,
     increment_incompatibility_count(readerStatus,
                                     OpenDDS::TRANSPORTTYPE_QOS_POLICY_ID);
   }
-
+  bool encapsulated_only = false;
+  bool exit_loop = false;
+  for (CORBA::ULong i = 0; i < pubTLS.length() && !exit_loop; ++i) {
+    for (CORBA::ULong j = 0; j < subTLS.length() && !exit_loop; ++j) {
+      if (0 == std::strcmp(pubTLS[i].transport_type, subTLS[j].transport_type)) {
+        if (0 == std::strcmp(pubTLS[i].transport_type, "rtps_udp")) {
```

Comment:
This should not have a dependency on the transport directly in the DCPS_Utils.  Since the `DataRepresentationIdSeq` for each side is known, is that not sufficient to determine compatibility?  At this point we're not dealing with the "implied defaults" of an empty sequence, we should know the actual contents of `DataRepresentationIdSeq`.